docs/topics/multiplatform/multiplatform-discover-project.md

```diff
@@ -1,169 +1,315 @@
-            }
+        commonMain {
+            // configure the commonMain source set
         }
-        val jvmMain by getting
-        val jvmTest by getting
-        val jsMain by getting
-        val jsTest by getting
-        val nativeMain by getting
-        val nativeTest by getting
     }
 }
 ```
 
-</tab>
-<tab title="Groovy" group-key="groovy">
+Source sets can be either platform-specific or intermediate.
 
-```groovy
-kotlin {
-    sourceSets {
-        commonMain {
+### Platform-specific source sets
 
-        }
-        commonTest {
-            dependencies {
-                implementation kotlin('test')
-            }
-        }
-        jvmMain {
+While having only common code is convenient, it's not always possible. Code in `commonMain` compiles to all declared
+targets, and Kotlin doesn't allow you to use any platform-specific APIs there.
 
-        }
-        jvmTest {
+In a multiplatform project with native and JS targets, the following code in `commonMain` doesn't compile:
 
-        }
-        jsMain {
+```kotlin
+// commonMain/kotlin/common.kt
+// Doesn't compile in common code
+fun common() {
+    java.io.File("greeting.txt").writeText("Hello, Multiplatform!")
+}
+```
 
-        }
-        jsTest {
+As a solution, Kotlin creates platform-specific source sets, also referred to as platform source sets. Each target has a
+corresponding platform source set that compiles for only that target. For example, a `jvm` target has the
+corresponding `jvmMain` source set that compiles to only the JVM. Kotlin allows using platform-specific dependencies in
+these source sets, for instance, JDK in `jvmMain`:
 
-        }
-        nativeMain {
+```kotlin
+// jvmMain/kotlin/jvm.kt
+// You can use Java dependencies in the `jvmMain` source set
+fun jvmGreeting() {
+    java.io.File("greeting.txt").writeText("Hello, Multiplatform!")
+}
+```
 
-        }
-        nativeTest {
+### Compilation to a specific target
 
-        }
-    }
+Compilation to a specific target works with multiple source sets. When Kotlin compiles a multiplatform project to a
+specific target, it collects all source sets labeled with this target and produces binaries from them.
+
+Consider an example with `jvm`, `iosArm64`, and `js` targets. Kotlin creates the `commonMain` source set for common code
+and the corresponding `jvmMain`, `iosArm64Main`, and `jsMain` source sets for specific targets:
+
+![Compilation to a specific target](specific-target-diagram.svg){width=700}
+
+During compilation to the JVM, Kotlin selects all source sets labeled with "JVM", namely, `jvmMain` and `commonMain`. It
+then compiles them together to the JVM class files:
+
+![Compilation to JVM](compilation-jvm-diagram.svg){width=700}
+
+Because Kotlin compiles `commonMain` and `jvmMain` together, the resulting binaries contain declarations from
+both `commonMain` and `jvmMain`. This is how you can choose which parts of your application should be shared.
```

Comment:
> This is how you can choose which parts of your application should be shared.

One more "bug" missed during the initial Google Doc review (sorry!): "this is how" is totally not clear. I think we can remove this sentence without any harm